File: 104. Maximum Depth of Binary Tree/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	return dfs(root, 0)
}

func dfs(node *TreeNode, depth int) int {
	if node == nil {
		return depth
	}
	return max(dfs(node.Left, depth+1), dfs(node.Right, depth+1))
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 104. Maximum Depth of Binary Tree/main.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.dfs(root, 0)

    def dfs(self, root, depth):
        if root is None:
            return depth
        return max(self.dfs(root.left, depth+1), self.dfs(root.right, depth+1))
